dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -625,6 +637,140 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
+        return Response.ok(bundleFile, "application/x-tgz").build();
+    }
+    
+    @Path("/_generate")
+    @POST
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.APPLICATION_JSON)
+    public final Response generateBundle(@Context final HttpServletRequest request,
+                                       @Context final HttpServletResponse response,
+                                       final DownloadBundleForm form) throws DotPublisherException, DotDataException {
+
+
+        User user = new WebResource.InitBuilder(request, response)
+                        .requiredBackendUser(true)
+                        .rejectWhenNoUser(true)
+                        .requiredPortlet("publishing-queue")
+                        .init()
+                        .getUser();
+
+
+
+        try {
+
+            final Bundle bundle = Try.of(()->APILocator.getBundleAPI().getBundleById(form.bundleId)).getOrElseThrow(e->new DotRuntimeException(e));
+            
+            
+            final String bundleName = bundle.getName().replaceAll("[^\\w.-]", "_");
+            
+            //set Filter to the bundle
+            bundle.setFilterKey(form.filterKey);
+            bundle.setOwner(user.getUserId());
+            //set ForcePush value of the filter to the bundle
+            bundle.setForcePush(
+                    (boolean) APILocator.getPublisherAPI().getFilterDescriptorByKey(form.filterKey).getFilters().getOrDefault(FilterDescriptor.FORCE_PUSH_KEY,false));
+            bundle.setOperation(form.operation.ordinal());
+            //Update Bundle
+            APILocator.getBundleAPI().updateBundle(bundle);
+
+            //Generate the bundle file for this given operation
+
+            BundleGenerator generator = new BundleGenerator(bundle, user);
+            
+            DotConcurrentFactory.getInstance().getSubmitter().submit(generator);
+            response.setContentType( "application/x-tgz" );
+            response.setHeader( "Content-Disposition", "attachment; filename=" +bundleName  +"-"+ bundle.getId() + ".tar.gz" );
+            final byte whiteByte = 0x20;
+            
+            final long nowsers = System.currentTimeMillis();
+            
+            // 10 secs
+            final int sleep = 1000;
+
+            // 2 hours
+            int loop = sleep * 60 * 120;
+            try(final OutputStream outStream = response.getOutputStream()){
+                while(generator.bundleFile==null) {
+                    Logger.infoEvery(this.getClass(), "writing bundleId:" + bundle.getId() + " time elapsed:" + Duration.ofMillis(System.currentTimeMillis()-nowsers).toString().substring(2).replaceAll("(\\d[HMS])(?!$)", "$1 ").toLowerCase(), 10000);
+                    outStream.write(whiteByte);
+                    outStream.flush();
+                    Try.run(()->Thread.sleep(sleep));
+                    if(--loop<=0)break;
```

Comment:
Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)